akka-remote/src/main/scala/akka/remote/artery/Association.scala

```diff
@@ -621,7 +622,12 @@ private[remote] class AssociationRegistry(createAssociation: Address ⇒ Associa
   private[this] val associationsByAddress = new AtomicReference[Map[Address, Association]](Map.empty)
   private[this] val associationsByUid = new AtomicReference[ImmutableLongMap[Association]](ImmutableLongMap.empty)
 
-  @tailrec final def association(remoteAddress: Address): Association = {
+  final def association(remoteAddress: Address): Option[Association] = {
+    val currentMap = associationsByAddress.get
+    currentMap.get(remoteAddress)
+  }
```

Comment:
Ahaaa, now I see what you meant with the stack unfolding from run*Stream. Ok, got it!